ocaml/lambda/lambda.mli

```diff
@@ -192,7 +192,10 @@ and array_kind =
 
 and value_kind =
     Pgenval | Pfloatval | Pboxedintval of boxed_integer | Pintval
-  | Pblock of { tag : int; fields : value_kind list }
+  | Pvariant of {
+      consts : int list;
+      non_consts : (int * value_kind list) list;
+    }
```

Comment:
Does this overlap with `Pintval` when `non_consts` is empty? Have we checked that everywhere which matches on `Pintval` also work for this?